helix-term/src/ui/editor.rs

```diff
@@ -1062,6 +1062,16 @@ impl Component for EditorView {
                     (Mode::Insert, Mode::Normal) => {
                         // if exiting insert mode, remove completion
                         self.completion = None;
+
+                        // For user friendly,
+                        // If we go from insert mode to normal node, and press no extra keys,
+                        // should try to make current line empty(restore pre-inserted indent).
+                        //
+                        // When we press no extra keys to normal mode, the `last_insert.1` just contains one key(ESC).
+                        if self.last_insert.1.len() == 1 {
+                            commands::Command::goto_line_start.execute(&mut cxt);
+                            commands::Command::kill_to_line_end.execute(&mut cxt);
```

Comment:
Why not just `kill_to_line_start`?

Reply:
I found when I put the cursor at the end of line, `kill_to_line_start` will kill the whole line, and we just need to make the current line empty.

I don't know if it's expected behavior, so I leave it unchange and use `goto_line_start` + `kill_to_line_end`.

Reply:
Is this possibly because `line_end_char_index` is the index of the new line, and thus is including the new line? Could you just go one character to the left of the end of the line?

Reply:
Thanks for your suggestion, but what if I'm editing a buffer with no indent specified(like scratch)?

If I simply go one character to the left, the cursor will move to the previous line.  Then execute `kill_to_line_end` will kill the previous line.